libs/s25main/buildings/noBuildingSite.h

```diff
@@ -69,27 +69,16 @@ class noBuildingSite : public noBaseBuilding
 
     ~noBuildingSite() override;
 
-    /// Aufräummethoden
-protected:
-    void Destroy_noBuildingSite();
+    void Destroy() override;
+    void Serialize(SerializedGameData& sgd) const override;
 
-public:
-    void Destroy() override { Destroy_noBuildingSite(); }
-
-    /// Serialisierungsfunktionen
-protected:
-    void Serialize_noBuildingSite(SerializedGameData& sgd) const;
-
-public:
-    void Serialize(SerializedGameData& sgd) const override { Serialize_noBuildingSite(sgd); }
-
-    GO_Type GetGOT() const override { return GO_Type::Buildingsite; }
+    GO_Type GetGOT() const final { return GO_Type::Buildingsite; }
```

Comment:
we could also make a template out of the base class and give it that enum on definition?

Reply:
That would mean we need a template which increases compile times.
Also it is not really useful here: This is exactly what a pure virtual method is for, so what would we gain? Even worse: We have this quite some way down the hierarchy chain